client/src/components/ChefProfile/ChefRecipes.js

```diff
@@ -0,0 +1,141 @@
+import React, { useContext, useState, useEffect } from "react";
+import { Box, Grid, Typography, Chip, Divider } from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import { RecipeContext } from "../../context/recipe-context";
+import { RecipeDispatchContext } from "../../context/recipe-context";
+import { setSrcDataToRecipe } from "../../actions/recipeActions";
+import plate from "../../assets/plate.svg";
+import useGetSrcData from "../../hooks/useGetSrcData";
+import DialogControl from "../Dialogs/DialogControl";
+
+const ChefProfile = ({ id }) => {
+  const { recipes } = useContext(RecipeContext);
+  const recipe = recipes.filter((res) => res._id === id)[0];
+  const {
+    name,
+    price,
+    ingredients,
+    requiredStuff,
+    portionDescription,
+    cuisineTags,
+    pictureKey,
+    srcData,
+  } = recipe;
+
+  const dispatch = useContext(RecipeDispatchContext);
+
+  const [src, setSrc] = useState(srcData ? srcData : null);
+  const [open, setOpen] = useState(false);
+
+  const getSrcData = useGetSrcData();
+  useEffect(() => {
+    const getImageSrcData = async () => {
+      if (srcData || !pictureKey) return;
+      const response = await getSrcData(pictureKey);
+      if (response.srcData) {
+        // setSrc(response.srcData);
+        setSrcDataToRecipe(dispatch, recipe._id, response.srcData);
+      }
+    };
+    if (srcData) {
+      setSrc(srcData);
+    }
+
+    getImageSrcData();
+  }, [pictureKey, getSrcData, srcData, dispatch, recipe._id]);
+
+  const handleClickOpen = (e) => {
+    setOpen(true);
+  };
+
+  const handleClose = (value) => {
+    setOpen(false);
+  };
+
+  // const {
+  //   name,
+  //   pictureUrl,
+  //   price,
+  //   ingredients,
+  //   requiredStuffs,
+  //   portionDescription,
+  //   cuisineTags,
+  // } = props.recipe;
```

Comment:
If this commented out code isn't needed, you can probably remove it and include it when it's needed